src/app/data-model-provider/MetadataTypes.h

```diff
@@ -75,6 +87,16 @@ struct ClusterInfo
     ClusterInfo(DataVersion version) : dataVersion(version) {}
 };
 
+// Constants used to narrow binary expressions
+constexpr uint8_t kPrivilegeBits   = 5;
```

Comment:
Please add unit tests here: add a unit test that all possible privileges and attribute qualities and command qualities can be encoded and successfully pass a `HAS` check.